src/segment/urlType.js

```diff
@@ -34,13 +35,33 @@ export const urlTypeToSegment = ({ baseUrl = '', source = '', range = '', indexR
   if (range || indexRange) {
     const rangeStr = range ? range : indexRange;
     const ranges = rangeStr.split('-');
-    const startRange = parseInt(ranges[0], 10);
-    const endRange = parseInt(ranges[1], 10);
+    let startRange = parseInt(ranges[0], 10);
```

Comment:
Unless I'm missing something, it seems like `parseInt` doesn't return a `BigInt`. We probably need to parse as `BigInt` and if it's under `MAX_SAFE_INTEREGER` convert it to a regular number.

Reply:
fixed